esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -1905,6 +1905,17 @@ void CWsDfuEx::doGetFileDetails(IEspContext &context, IUserDescriptor* udesc, co
     if (version >= 1.28)
         FileDetails.setNumParts(df->numParts());
 
+    if (cluster && *cluster && (version >= 1.31))
+    {
+        Owned<IFileDescriptor> fdesc = df->getFileDescriptor(cluster);
```

Comment:
I would not bother with providing cluster here.. as you are then going to ask for the specific IClusterInfo by cluster below anyway.